bson/unmarshal_test.go

```diff
@@ -95,18 +95,105 @@ func TestUnmarshalExtJSONWithRegistry(t *testing.T) {
 }
 
 func TestUnmarshalExtJSONWithContext(t *testing.T) {
-	t.Run("UnmarshalExtJSONWithContext", func(t *testing.T) {
-		type teststruct struct{ Foo int }
-		var got teststruct
-		data := []byte("{\"foo\":1}")
-		dc := bsoncodec.DecodeContext{Registry: DefaultRegistry}
-		err := UnmarshalExtJSONWithContext(dc, data, true, &got)
-		noerr(t, err)
-		want := teststruct{1}
-		if !cmp.Equal(got, want) {
-			t.Errorf("Did not unmarshal as expected. got %v; want %v", got, want)
-		}
-	})
+	var cases = []struct {
+		name  string
+		sType reflect.Type
+		want  interface{}
+		data  []byte
+	}{
+		{
+			name: "Small struct",
+			sType: reflect.TypeOf(struct {
+				Foo int
+			}{}),
+			data: []byte(`{"foo":1}`),
+			want: &struct {
+				Foo int
+			}{Foo: 1},
+		},
+		{
+			name: "Valid surrogate pair",
+			sType: reflect.TypeOf(struct {
```

Comment:
nit: it may be worth declaring this struct if it's used in this many tests

Reply:
Good idea, updated.